### ISSUE TITLE
SQLite 3.19.3 breaks EF query

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#8904]
SQLitePCLRaw 1.1.6 includes SQLite 3.19.3. SQLitePCLRaw 1.1.3 included SQLite 3.16.0. Looks like the SQLite update includes changes to JOINs that may affect query results:
http://sqlite.org/releaselog/3_19_3.html

See https://github.com/aspnet/EntityFramework/pull/8890#issuecomment-309579098
And http://sqlite.org/releaselog/3_19_3.html

**Comment by ajcvickers** [comment]
@anpete @smitpatel @maumar Could one of you investigate this please?

/cc @bricelam @divega @natemcmaster @Eilon

**Comment by ajcvickers** [comment]
@natemcmaster Do we need this for preview2?

**Comment by natemcmaster** [comment]
No, this can wait until RTM.

**Comment by ericsink** [comment]
FWIW, I can reproduce this by building the EF core tree and running the tests.  As indicated by others, the test pass with SQLitePCL.raw 1.1.3, and 2 of them fail with SQLitePCL.raw 1.1.6.

For diagnostic purposes, I created a SQLitePCL.raw version 1.1.7-pre20170619212538 which uses SQLite 3.18.0 in the Windows builds.  With this version substituted into build/dependencies.props, the tests once again pass.

So the problem slipped into SQLite between 3.18.0 and 3.19.3.  Given the nature of the failing test cases, it seems like it's gotta be related to the LEFT JOIN changes mentioned in the 3.19 release notes.

**Comment by ericsink** [comment]
Update:  It is easy to see the problem with just the sqlite3 shell.  

I grabbed the InheritanceRelationshipsQuerySqliteTest.db file (from test/EFCore.Sqlite.FunctionalTests/bin/Debug/netcoreapp2.0) and a query from one of the failing tests.

This makes for a nice, self-contained repro.  Running it with 3.19.3 gives different results than 3.18.0.  

So I sent a message to the sqlite developers.  I'll post another update here after I get more info.

**Comment by ericsink** [comment]
SQLite ticket:

http://www.sqlite.org/src/tktview/892fc34f173e99d8aa03d52fbf42e0f9f2dfb210

**Comment by natemcmaster** [comment]
@ericsink Thanks for helping investigate this. Are you going to rollback SQLitePCLRaw to 3.18 or wait for a fix from SQLite?

**Comment by anpete** [comment]
Thanks @ericsink!

**Comment by ericsink** [comment]
I plan to wait for some indication on the timing of a fix before deciding that.  DRH is usually very fast on this kind of thing.  Then again, that particular code change in 3.19.0 has now been the cause of 3 sqlite bugs, and that kind of regression is extremely unusual for sqlite, so the circumstances are atypical.

I don't consider 3.18.0 to be an acceptable option, because it has a couple of other known issues.  I only used it for my experiment because the more-patched 3.18.2 was not available as a convenient amalgamation.

TLDR:  I will either go with 3.18.2 or wait for the fix in (what I presume will be) 3.19.4.

**Comment by ericsink** [comment]
@natemcmaster You asked: "Are you going to rollback SQLitePCLRaw to 3.18 or wait for a fix from SQLite?"

Do you have a preference?

I've done the prep work for a release with SQLite 3.18.2.  And I've confirmed that it passes the EF Core tests.

I could push this as 1.1.7 to nuget tomorrow, but I'm on the fence about whether I should proceed that way or just wait for a 3.19.4.

**Comment by natemcmaster** [comment]
I'm on the fence too. I see DRH already committed a fix in SQLite. http://www.sqlite.org/src/artifact/879fcac86a74ea1a. If we can expect SQLite 3.19.4 in the next week or two, we're probably ok to wait. But if not, 3.18.2 would be better.

**Comment by ericsink** [comment]
I'll wait, for now.  I can pull the trigger any time if it becomes more urgent.  But I expect 3.19.4 will probably happen soon.

**Comment by ajcvickers** [comment]
Moving out of milestone so we can triage.

**Comment by ajcvickers** [comment]
Closing this as there is nothing actionable on EF. We are happy waiting as discussed above.

@natemcmaster Are we tracking removal of the package workarounds somewhere else?

**Comment by natemcmaster** [comment]
@ajcvickers tracking it here: https://github.com/aspnet/External/issues/59

**Comment by natemcmaster** [comment]
> I'll wait, for now. I can pull the trigger any time if it becomes more urgent.

@ericsink can we move forward with a 1.1.7 update that rollsback to SQLite 3.18.0? Our timelines have shifted and it would be good to get this resolved sooner as we don't know when DRH will get around to 3.19.4.

**Comment by ericsink** [comment]
Yep.  I'll do it today.

Minor nit:  I plan to include 3.18.2, not 3.18.0.  I assume that's okay.

**Comment by natemcmaster** [comment]
Thanks. And yes, 3.18.2 is fine.

**Comment by ericsink** [comment]
@natemcmaster SQLitePCLRaw 1.1.7 has been pushed to nuget.  It appears to pass everything in EFCore.Sqlite.FunctionalTests.

**Comment by natemcmaster** [comment]
Great, I'll prepare the appropriate PRs to update.

**Comment by natemcmaster** [comment]
Ok. 1.1.7 is in. Thanks again Eric.
